### PR TITLE
Big various updates

### LINKED ISSUE #15: Add README.md for every npm published packages

### LINKED ISSUE #18: CSV loader
Now done with:

`yarn add -D csv-loader`

and in `webpack.config.js`:

```
const buildWebpackConfig = require('webpack-preset-accurapp')
module.exports = buildWebpackConfig([
  (context) => ({
    module: {
      loaders: [
        {
          test: /\.csv$/,
          loader: 'csv-loader',
          options: {
            dynamicTyping: true,
            header: true,
            skipEmptyLines: true,
          },
        },
      ],
    },
  }),
])
```

### LINKED ISSUE #19: SVG Loader
https://github.com/boopathi/react-svg-loader

### LINKED ISSUE #22: Remove async/await from babelrc in development
Refactor the babelrc when babel7 comes out.

Also add babel-macros 
https://babeljs.io/blog/2017/09/11/zero-config-with-babel-macros

And use browserlist
https://github.com/babel/babel-preset-env/issues/149

### LINKED ISSUE #23: 🚨 Upgrade eslint to v4
https://github.com/eslint/eslint.github.io/blob/master/docs/4.0.0/user-guide/migrating-to-4.0.0.md

Also check out 

https://github.com/buildo/eslint-config

### LINKED ISSUE #24: Make devServer options completely configurable
Every property should be overridable either in `package.json` or in `webpack.config.js`

### LINKED ISSUE #30: Replace lerna with yarn workspaces
https://yarnpkg.com/blog/2017/08/02/introducing-workspaces/

Lerna will still handle the publishing.

[WIP branch is here](https://github.com/accurat/accurapp/tree/yarn-workspaces)

Currently waiting on [this issue](https://github.com/yarnpkg/yarn/issues/4469) to be resolved

### LINKED ISSUE #32: In production, transpile the entire bundle to prevent leaking ES6 in production
Should be enough to remove `exclude: 'node_modules'` from the babel config of the prod build configuration.

### LINKED ISSUE #33: Don't check for newer accurapp versions during CI build

### LINKED ISSUE #36: Replace accurapp logo with a smaller one
Reduce the used logo to a more minimal one so that with small console is readable.
Maybe changing the ASCII art font or just using the /|||/| logo.

### PR DESCRIPTION
**Pull request by marcofugaro**
## New features
- possibility to edit the babel configuration via the `.babelrc`
- [babel-plugin-macros](https://github.com/kentcdodds/babel-plugin-macros), [pipeline-operator](https://github.com/babel/babel/tree/master/packages/babel-plugin-proposal-pipeline-operator), [optional-chaining](https://github.com/babel/babel/tree/master/packages/babel-plugin-proposal-optional-chaining), [optional-catch-binding](https://github.com/babel/babel/tree/master/packages/babel-plugin-proposal-optional-catch-binding) and many other from babel7
- transpilation of the `node_modules`
- new Eslint rules (already published)
- possiblity to edit the supported browsers both in development and in production via `browserslist` in `package.json`. Both babel-preset-env and autoprefixer depend on it.
- CSV Loader
- CSS Modules for `*.module.css` files
- Support for [FUSS](https://github.com/caesarsol/postcss-fuss)
- CSS Nesting
- [JSON5](https://json5.org/) Loader
- support for the `.env` file
- A lot of **documentation**, including:
  - Netlify setup guide and config file
  - Adding new webpack loaders and plugins
  - Adding support for typescript
  - Available Env variables
  - How do I enable hot reloading?
  - Where do I put the images?
  - Where do I put the custom fonts?
  - What is the public folder for?
  - How do I override a webpack loader?
- Fix forever the PUBLIC_URL env
- Add the LATEST_COMMIT and LATEST_COMMIT_TIMESTAMP env variables
- Add support for `color-mod()` in css using postcss
- Lint command
- Prettier command
- Styles are now imported from the <head> avoiding the FOUC
- SVG React Loader

## Refactoring
- Updated the `create-test-app` command according to the new yarn version
- Updated lerna and the publish command
- remove BSD oudated license
- add short README in every package
- updated the start and build node scripts according to the updates in create-react-app
- upgrade deps to webpack4, babel7, eslint5
- deprecated setting the NODE_ENV from the outside, other other variables instead
- updated the webpack.config

Closes #15
Closes #18
Closes #19 
Closes #22
Closes #23
Closes #24
Closes #30
Closes #32
Closes #33
Closes #36

### CODE REVIEW

**Comment by caesarsol** [comment]
HOLY SHEET!!!!!!!

File: packages/create-accurapp/index.js [hunk 1]
```diff
@@ -123,6 +123,10 @@ const packageJson = {
     start: 'accurapp-scripts start',
     build: 'accurapp-scripts build',
   },
+  browserslist: {
+    production: ['last 2 versions', 'ie 11'],
```
Comment (markov00) [inline packages/create-accurapp/index.js]: Take a read over this blog post: https://jamie.build/last-2-versions

File: packages/webpack-preset-accurapp/package.json [hunk 2]
```diff
@@ -5,11 +5,11 @@
   "repository": "accurat/accurapp",
   "license": "MIT",
   "engines": {
-    "node": ">=6.5"
+    "node": ">=7"
   },
-  "keywords": [
```
Comment (markov00) [inline packages/webpack-preset-accurapp/package.json]: I think we can keep the keywords list, maybe adding some more interesting keyword like
`preset`,`webpack`,`accurat`

File: packages/create-accurapp/index.js (hunk 1 above)
Comment (marcofugaro) [reply packages/create-accurapp/index.js]: @markov00 already done, check the Files Changed not the commits please. This is outdated

File: packages/webpack-preset-accurapp/package.json (hunk 2 above)
Comment (marcofugaro) [reply packages/webpack-preset-accurapp/package.json]: ci sta, fatto

**Comment by caesarsol** [comment]
@accurat/developers I propose: before merging, can we publish it in a `-beta-1` version and try it out on some bigger projects?

**Comment by marcofugaro** [comment]
good job @marcofugaro 🎉

**Pull request merged**